Sources/MapboxNavigation/NavigationComponent.swift

```diff
@@ -15,6 +16,11 @@ public protocol NavigationMapInteractionObserver: AnyObject {
      Called when the NavigationMapView centers on a location.
      */
     func navigationViewController(didCenterOn location: CLLocation)
+    
+    /**
+     Called when the NavigationMapView updates on style.
+     */
+    func navigationViewController(updateTo styleURI: StyleURI?)
```

Comment:
Added the method to notify the style change, and to boost the `SpriteRepository` update within the components with new style and downloading process.